backend/src/main/java/com/github/lenaruppert/backend/controller/MongoUserController.java

```diff
@@ -0,0 +1,102 @@
+package com.github.lenaruppert.backend.controller;
+
+import com.github.lenaruppert.backend.model.MongoUser;
+import com.github.lenaruppert.backend.model.MongoUserDTO;
+import com.github.lenaruppert.backend.repository.MongoUserRepository;
+import jakarta.servlet.http.HttpSession;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.security.Principal;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("api/user")
+@RequiredArgsConstructor
+public class MongoUserController {
+
+    private final MongoUserRepository mongoUserRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    @PostMapping
+    public MongoUser create(@RequestBody MongoUserDTO user) {
+        if (user.username() == null || user.username().length() == 0) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Username is required");
+        }
+
+        if (user.password() == null || user.password().length() == 0) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Password is required");
+        }
+
+        if (mongoUserRepository.existsByUsername(user.username())) {
+            throw new ResponseStatusException(
+                    HttpStatus.CONFLICT,
+                    "User already exists"
+            );
+        }
+
+        MongoUser newUser = new MongoUser(
+                UUID.randomUUID().toString(),
```

Comment:
It is normally advised to implement an `IdService` for UUID generation, makes it easier to test everything :)